### PR TITLE
Fixes #31307 - Improve compute network error messages

### PR DESCRIPTION
**Pull request by sbernhard**
Previously, if the connection to the compute resource fails, the
following error message was shown.
  Failed to find compute attributes, please check if VM 'xyz' was deleted.

Users are completely confused. With the following change nice error
messages are shown:

Reject:
  Receiving vm data for host 'xyz.local' from used compute resource 'VMware' failed: 'Connection to compute resource refused'.

Unreachable:
  Receiving vm data for host 'xyz.local' from used compute resource 'VMware' failed: 'Compute resource destination unreachable'.

Timeout:
  Receiving vm data for host 'xyz.local' from used compute resource 'VMware' failed: 'Connection to compute resource timed out'.

### CODE REVIEW

**Comment by sbernhard** [comment]
For testing: 
```
# iptables --append OUTPUT --protocol tcp -d 192.168.10.1 --jump  REJECT
# hammer host update --id 9 --parameters t3=t6
Could not update the host:
  Receiving vm data for host 'xyz.local' from used compute resource 'VMware' failed: 'Connection to compute resource refused'.

# iptables --append OUTPUT --protocol tcp -d 192.168.10.1 --jump  REJECT --reject-with icmp-host-unreachable
# hammer host update --id 9 --parameters t3=t6
Could not update the host:
  Receiving vm data for host 'xyz.local' from used compute resource 'VMware' failed: 'Compute resource destination unreachable'.

# iptables --append OUTPUT --protocol tcp -d 192.168.10.1 --jump DROP
# hammer host update --id 9 --parameters t3=t6
Could not update the host:
  Receiving vm data for host 'xyz.local' from used compute resource 'VMware' failed: 'Connection to compute resource timed out'.
```

**Comment by sbernhard** [comment]
I don't think that the foreman test issue in 2.7 is related to this change.

File: app/models/concerns/orchestration/compute.rb [hunk 1]
```diff
@@ -17,7 +17,19 @@ def compute?
...
+        compute_resource.find_vm_by_uuid(uuid)
+      rescue Net::OpenTimeout, Errno::ECONNREFUSED, Errno::EHOSTUNREACH => e
+        failure _("Receiving vm data for host '%{host}' from used compute resource '%{compute_resource}' failed: '%{exception}'.") % {
+          :host => vm_name,
+          :compute_resource => compute_resource.to_s,
+          :exception => compute_resource_connection_failure(e),
+        }
+        nil
+      rescue => e
+        logger.debug "Receiving vm data for host '#{vm_name}' from compute resource '#{compute_resource}' failed. #{e.class}: #{e.message}"
```
Comment (timogoebel) [inline app/models/concerns/orchestration/compute.rb:30]:
```suggestion
        Foreman::Logging.exception("Receiving vm data for host '#{vm_name}' from compute resource '#{compute_resource}' failed.", e)
```

**Review by lzap — DISMISSED** [review]
I am good, @timogoebel ?

**Review by timogoebel — DISMISSED** [review]
@sbernhard: I'm good as well, please make sure the checks are green or else we can't merge.

**Review by timogoebel — DISMISSED** [review]

**Comment by ekohl** [comment]
On merge we should fix up the commit message to match what I did to the title of this PR. It's just a bit more descriptive.

**Comment by sbernhard** [comment]
> On merge we should fix up the commit message to match what I did to the title of this PR. It's just a bit more descriptive.

done. Thanks @ekohl

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
[test upgrade]

**Comment by timogoebel** [comment]
@ezr-ondrej: Are you good as well?

**Review by ezr-ondrej — APPROVED** [review]
> @ezr-ondrej: Are you good as well?

Yeah, I just didn't see you've already reviewed :)

**Comment by timogoebel** [comment]
Thanks, everyone! :)

> Yeah, I just didn't see you've already reviewed :)

I need to gain some karma points today. :)

**Comment by tbrisker** [comment]
Cherry pick in GH-8616

**Pull request merged**